### PR TITLE
PP-9334: add new table for payment instruments and foreign keys to charges and agreements tables

### PR DESCRIPTION
**Pull request by gidsg**
Co-authored-by: Alan Carter <alan.carter@digital.cabinet-office.gov.uk>

## WHAT YOU DID
Add database migration

## How to test

Check code makes sense, migrations should run successfully when app is run locally

### Logging

- [ ] only emit log lines at ERROR level which require immediate attention from a support engineer. These will trigger a zendesk alert.

### Documentation

- [ ] Updated README.md for any of the following ?

* Introduced any new environment variables / removed existing environment variable
* Added new API / updated existing API definition

### CODE REVIEW

File: src/main/resources/migrations.xml [hunk 1]
```diff
@@ -1776,5 +1776,80 @@
...
+                <constraints nullable="true"/>
+            </column>
+            <column name="status" type="varchar(255)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="card_brand" type="text">
+                <constraints nullable="true"/>
+            </column>
+            <column name="start_date" type="timestamp without timezone">
+                <constraints nullable="true"/>
```
Comment (alexbishop1) [inline src/main/resources/migrations.xml:1826]: Will we ever have a payment instrument without a start date?

File: src/main/resources/migrations.xml [hunk 2]
```diff
@@ -1776,5 +1776,80 @@
...
+            <column name="recurring_auth_token" type="jsonb">
+                <constraints nullable="true"/>
+            </column>
+        </createTable>
+
+    </changeSet>
+
+    <changeSet id="add payment_instrument_id to charges" author="">
+        <addColumn tableName="charges">
+            <column name="payment_instrument_id" type="bigserial">
```
Comment (alexbishop1) [inline src/main/resources/migrations.xml]: I think this should be `bigint` (although the column referenced by the foreign key constraint auto-increments, _this_ column does not).

File: src/main/resources/migrations.xml [hunk 3]
```diff
@@ -1776,5 +1776,80 @@
...
+                <constraints foreignKeyName="fk__charges_payment_instruments"
+                             referencedTableName="payment_instruments"
+                             referencedColumnNames="id" nullable="true"/>
+            </column>
+        </addColumn>
+    </changeSet>
+
+    <changeSet id="add payment_instrument_id to agreements" author="">
+        <addColumn tableName="agreements">
+            <column name="payment_instrument_id" type="bigserial">
```
Comment (alexbishop1) [inline src/main/resources/migrations.xml]: `bigint`

**Review by alexbishop1 — CHANGES_REQUESTED** [review]

File: src/main/resources/migrations.xml (hunk 1 above)
Comment (gidsg) [reply src/main/resources/migrations.xml:1826]: Could there in practice be any difference between the `start_date` and `created_date` i.e. could the payment instrument ever be added and the start date added later? if not could we remove it, or default it to `now()` like `created_date`?

Comment (alexbishop1) [reply src/main/resources/migrations.xml:1826]: Let’s leave it for now; we can always change it later.

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**